src/core/include/units/quantity.h

```diff
@@ -465,6 +465,17 @@ namespace detail {
 template<typename D, typename U, typename Rep>
 inline constexpr bool is_quantity<quantity<D, U, Rep>> = true;
 
+template<typename T>
+requires units::is_derived_from_specialization_of<T, units::quantity> &&
+        requires {
+    typename T::dimension;
+    typename T::unit;
+    typename T::rep;
+    requires Dimension<typename T::dimension>;
+    requires Unit<typename T::unit>;
+    requires Representation<typename T::rep>; }
+inline constexpr bool is_quantity<T> = true;
+
```

Comment:
If you want types derived from `quantity` to match the `Quantity` concept probably all that is needed is:

```cpp
template<typename T>
requires units::is_derived_from_specialization_of<T, units::quantity>
inline constexpr bool is_quantity<T> = true;
```

even removing the definition in lines 465-466.

However I am not sure if that solves your issue. See the next comment...

Reply:
The rationale behind checking for inheritance is that the template class that derives from 'quantity' can have various different additional signatures and not a single static one. The unit-tests does not convey this properly because there is just the addition of one string litteral.

Reply:
> even removing the definition in lines 465-466.

The rationale behind these was to make the concept checks for the existence of the 'using' as early as possible and also to ensure that `T' is `Representation`, `Dimension`, and `unit`. 

Our use-case also works without but I saw in other parts of units that these are checked for... if you could confirm that these lines should be removed.

Also: thanks a lot for your time in reviewing this PR! Much appreciated.

Reply:
I have to say I do not understand your answers.

What do you mean by:
> the template class that derives from 'quantity' can have various different additional signatures and not a single static one

Do you mean a multiple inheritance here? Still a `is_derived_from_specialization_of` should cover this. It covers the class that is the `quantity` class template and all children that publicly derive from it. If you publically derive from it then `dimension`, `unit`, and `rep` are always exposed unless you override them with a different type in a child class but I do not think we should check for such a case.

Reply:
@mpusz you are absolutely right. I mixed up the actual `is_derived_from_specialization_of<...>` with a similar version I saw/used elsewhere that had different internal workings.  Will remove the additional `requires` statement. Sorry for the confusion.

Reply:
incorporated requested change